cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/DataFlowUtil.qll

```diff
@@ -1947,7 +2065,12 @@ module ExprFlowCached {
     isIndirectBaseOfArrayAccess(n, result)
     or
     not isIndirectBaseOfArrayAccess(n, _) and
-    result = n.asExpr()
+    (
+      result = n.asExpr()
+      or
+      result = n.asDefinition() and
+      result instanceof CrementOperation
+    )
```

Comment:
Let me split the answer into two paragraphs:
### Why does this need updating?
It needed to be updated to match the use in the `CWE-119` queries (where the `Buffer.qll` library expect `localExprFlow(e1, e2)` to have a result when `e1` is a postfix increment expression. This worked before since `asExpr` conflated the value before the increment with the value after the increment, but now that this conflation is gone we need to do this conflation manually in the `asExpr` predicate used for computing the `localFlowExpr` relation.

### Why only for crement and not for the assignment store?

That's a very good catch! In ce28c9b485eefdf6d7cccada4fe4a30687cffde4 I've added a testcase that fails because I forgot this, and in 359b15bb60e0656fe614927728fb58900ffec8ac I've fixed the FP by adding the missing case.